DeviceAdministration/Infrastructure/Repository/DeviceTelemetryRepository.cs

```diff
@@ -263,45 +263,77 @@ public DeviceTelemetryRepository(IConfigurationProvider configProvider)
                         model.DeviceId = str;
                     }
 
-                    if (strdict.TryGetValue("externaltemperature", out str) &&
-                        double.TryParse(
-                            str,
-                            NumberStyles.Float,
+                    DateTime date;
+                    if (strdict.TryGetValue("eventenqueuedutctime", out str) &&
+                        DateTime.TryParse(
```

Comment:
If we're not handling the 'false' case of TryParse, we should allow it throw an error.